cmd/kubeadm/app/apis/kubeadm/validation/validation.go

```diff
@@ -146,10 +146,9 @@ func ValidateArgSelection(cfg *kubeadm.NodeConfiguration, fldPath *field.Path) f
 		allErrs = append(allErrs, field.Invalid(fldPath, "", "DiscoveryTokenCACertHashes cannot be used with DiscoveryFile"))
 	}
 
-	// TODO: convert this warning to an error after v1.8
-	if len(cfg.DiscoveryFile) == 0 && len(cfg.DiscoveryTokenCACertHashes) == 0 && !cfg.DiscoveryTokenUnsafeSkipCAVerification {
-		fmt.Println("[validation] WARNING: using token-based discovery without DiscoveryTokenCACertHashes can be unsafe (see https://kubernetes.io/docs/admin/kubeadm/#kubeadm-join).")
-		fmt.Println("[validation] WARNING: Pass --discovery-token-unsafe-skip-ca-verification to disable this warning. This warning will become an error in Kubernetes 1.9.")
+	if len(cfg.DiscoveryFile) == 0 && len(cfg.DiscoveryToken) != 0 && len(cfg.DiscoveryTokenCACertHashes) == 0 && !cfg.DiscoveryTokenUnsafeSkipCAVerification {
+		allErrs = append(allErrs, field.Invalid(fldPath, "",
```

Comment:
"discovery-token-unsafe-skip-ca-verification"

Reply:
Thanks @luxas! Will address this and other comments tomorrow :)

Reply:
@luxas other comments addressed. But I'm a bit confusing here now. What were you expecting here to be? `allErrs = append(allErrs, field.Invalid(fldPath, "discovery-token-unsafe-skip-ca-verification",` or something else?